src/components/CheckForm/FormLayout/FormSidebar.tsx

```diff
@@ -0,0 +1,66 @@
+import React from 'react';
+import { GrafanaTheme2 } from '@grafana/data';
+import { config } from '@grafana/runtime';
+import { Icon, useStyles2 } from '@grafana/ui';
+import { css, cx } from '@emotion/css';
+
+export function FormSidebar({
+  sections,
+  onSectionSelect,
+  activeIndex,
+}: {
+  sections: Array<{ label: string; hasErrors: boolean; required?: boolean }>;
+  onSectionSelect: (index: number) => void;
+  activeIndex: number;
+}) {
+  const styles = useStyles2(getStyles);
+  return (
+    <ol className={styles.container}>
+      {sections.map(({ label: sectionTitle, required, hasErrors }, sectionIndex) => {
+        const prefix = hasErrors ? (
+          <Icon name={`exclamation-triangle`} color={config.theme2.colors.error.main} />
+        ) : (
+          <span className={css({ width: '16px', display: 'inline-block' })}>{sectionIndex + 1}</span>
+        );
+        return (
+          <li
+            key={sectionTitle}
+            onClick={() => onSectionSelect(sectionIndex)}
```

Comment:
Nest a button inside the `li` and here and move the onClick to that so we get lots of a11y niceties 😃